### PR TITLE
feat(puppet): add manifest and module to install sumologic-otel-collector

### PR DESCRIPTION
**Pull request by kkujawa-sumo**

### CODE REVIEW

File: vagrant/provision.sh [hunk 1]
```diff
@@ -24,3 +24,44 @@ curl -LJ "https://golang.org/dl/go${GO_VERSION}.linux-amd64.tar.gz" -o go.linux-
...
+192.168.79.13 puppetserver.example.com puppet
+192.168.79.13 agent.example.com
+END
+
+# Install puppet server & puppet agent
+wget https://apt.puppetlabs.com/puppet6-release-focal.deb
+dpkg -i puppet6-release-focal.deb
+apt-get update -y
+apt-get install puppetserver -y
+apt-get install puppet-agent -y
```
Comment (pmalek-sumo) [inline vagrant/provision.sh:39]:
```suggestion
apt-get install puppetserver puppet-agent -y
```

Why not together? Should be a touch faster

File: vagrant/provision.sh [hunk 2]
```diff
@@ -24,3 +24,44 @@ curl -LJ "https://golang.org/dl/go${GO_VERSION}.linux-amd64.tar.gz" -o go.linux-
 
 # Install ansible
 pip3 install ansible
+
+# Add puppet hosts
+tee -a /etc/hosts << END
+192.168.79.13 puppetserver.example.com puppet
+192.168.79.13 agent.example.com
```
Comment (pmalek-sumo) [inline vagrant/provision.sh:31]: What's the purpose of those?

Comment (sumo-drosiek) [reply vagrant/provision.sh:31]: if it's essential, could we use `127.0.0.1`?

Comment (kkujawa-sumo) [reply vagrant/provision.sh]: Puppet agent and server communicate using hosts names, certificates for agent and server needs to be generated for different different host names to have the same behaviour as in communication with agent on different machine (in vagrant there is server and agent on the single host). 
I changed IP address to `127.0.0.1`.

Comment (sumo-drosiek) [reply vagrant/provision.sh]: I suppose using `localhost` and `sumologic-otel-collector` hostnames doesn't work?

Comment (kkujawa-sumo) [reply vagrant/provision.sh]: I changed puppet server to sumologic-otel-collector

**Review by sumo-drosiek — APPROVED** [review]

**Pull request merged**